src/spec-node/collectionCommonUtils/outdatedCommandImpl.ts

```diff
@@ -0,0 +1,441 @@
+import { DevContainerConfig, DevContainerFromDockerfileConfig, getDockerComposeFilePaths } from '../../spec-configuration/configuration';
+import { CommonParams, ManifestContainer, getRef, getVersionsStrictSorted } from '../../spec-configuration/containerCollectionsOCI';
+import { DockerCLIParameters } from '../../spec-shutdown/dockerUtils';
+import { request } from '../../spec-utils/httpRequest';
+import { readDockerComposeConfig, getBuildInfoForService } from '../dockerCompose';
+import { extractDockerfile, findImage } from '../dockerfileUtils';
+import { ContainerFeatureInternalParams, userFeaturesToArray, getFeatureIdType, DEVCONTAINER_FEATURE_FILE_NAME, Feature } from '../../spec-configuration/containerFeaturesConfiguration';
+import { readLockfile } from '../../spec-configuration/lockfile';
+
+export async function outdated({
+	// 'user-data-folder': persistedFolder,
+	'workspace-folder': workspaceFolderArg,
+	config: configParam,
+	'output-format': outputFormat,
+	'log-level': logLevel,
+	'log-format': logFormat,
+	'terminal-rows': terminalRows,
+	'terminal-columns': terminalColumns,
+}: OutdatedArgs) {
+	const disposables: (() => Promise<unknown> | undefined)[] = [];
+	const dispose = async () => {
+		await Promise.all(disposables.map(d => d()));
+	};
+	let output: Log | undefined;
+	try {
+		const workspaceFolder = path.resolve(process.cwd(), workspaceFolderArg);
+		const configFile = configParam ? URI.file(path.resolve(process.cwd(), configParam)) : undefined;
+		const cliHost = await getCLIHost(workspaceFolder, loadNativeModule, logFormat === 'text');
+		const extensionPath = path.join(__dirname, '..', '..');
+		const sessionStart = new Date();
+		const pkg = getPackageConfig();
+		output = createLog({
+			logLevel: mapLogLevel(logLevel),
+			logFormat,
+			log: text => process.stderr.write(text),
+			terminalDimensions: terminalColumns && terminalRows ? { columns: terminalColumns, rows: terminalRows } : undefined,
+		}, pkg, sessionStart, disposables);
+
+		const workspace = workspaceFromPath(cliHost.path, workspaceFolder);
+		const configPath = configFile ? configFile : await getDevContainerConfigPathIn(cliHost, workspace.configFolderPath);
+		const configs = configPath && await readDevContainerConfigFile(cliHost, workspace, configPath, true, output) || undefined;
+		if (!configs) {
+			throw new ContainerError({ description: `Dev container config (${uriToFsPath(configFile || getDefaultDevContainerConfigPath(cliHost, workspace!.configFolderPath), cliHost.platform)}) not found.` });
+		}
+
+		const cacheFolder = await getCacheFolder(cliHost);
+		const params = {
+			extensionPath,
+			cacheFolder,
+			cwd: cliHost.cwd,
+			output,
+			env: cliHost.env,
+			skipFeatureAutoMapping: false,
+			platform: cliHost.platform,
+		};
+
+		const outdatedFeatures = await loadFeatureVersionInfo(params, configs.config.config);
+
+		const outputParams = { output, env: process.env };
+		const dockerParams = await createDockerParams({
```

Comment:
We seem to be using only a very small part of this parameter object. Would it make sense to pass only the required properties around instead of having the full object?

Reply:
Looking at how `dockerParams` is used later, it would be sufficient to create and pass the simpler `DockerCLIParameters` around. readConfiguration also does that: https://github.com/devcontainers/cli/blob/ab79dd6508f446185bb8b958b042cc8379061eaf/src/spec-node/devContainersSpecCLI.ts#L1011C17-L1011C36

Reply:
Sweet, updated!